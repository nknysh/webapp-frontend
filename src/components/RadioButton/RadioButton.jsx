import React from 'react';

import { Container, Text } from './RadioButton.styles';

const RadioButton = ({ name, placeholder, value, onChange, onBlur, style }) => (
  <Container>
    <input type="radio" name={name} checked={value} onChange={onChange} />
    <Text style={style}>{placeholder}</Text>
  </Container>
);

export default RadioButton;
