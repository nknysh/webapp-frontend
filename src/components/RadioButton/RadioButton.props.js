import PropTypes from 'prop-types';

export const propTypes = {
  options: PropTypes.any,
};

export const defaultProps = {
  options: [],
};
