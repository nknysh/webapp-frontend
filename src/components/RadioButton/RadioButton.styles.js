import styled from 'styled-components';

import theme from 'styles/theme';

export const Container = styled.div`
  flex-direction: row;
  align-items: center;
`;

export const Text = styled.span`
  margin-left: 5px;
  color: ${theme.primary};
`;
