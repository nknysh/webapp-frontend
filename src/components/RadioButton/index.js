// Libraries
import React from 'react';

// App
import { colors } from 'styles';

// Components
import Styled from '../Styled';

const Container = Styled.View.extend`
  flex-direction: row;
  align-items: center;
`;

const Text = Styled.H8.extend`
  margin-left: 5px;
  color: ${colors.gold10};
`;

const RadioButton = ({ name, placeholder, value, onChange, onBlur, style }) => (
  <Container>
    <input
      type="radio"
      name={name}
      checked={value}
      onChange={onChange}
    />
    <Text
      style={style}
    >
      {placeholder}
    </Text>
  </Container>
);

export default RadioButton;
