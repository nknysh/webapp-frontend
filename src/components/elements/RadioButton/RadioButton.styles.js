import styled from 'styled-components';
import { RadioGroup, FormControlLabel, Radio } from '@material-ui/core';

import theme from 'styles/theme';

export const StyledRadioGroup = styled(RadioGroup)`
  flex-direction: row !important;
`;

export const RadioFormControl = styled(FormControlLabel)`
  flex: 1 1 100%;
  text-transform: uppercase;
  margin-bottom: ${theme.gutter}px;

  > span {
    font-size: ${theme.fonts.sizes.normal}px !important;
    font-family: ${theme.fonts.defaultFont} !important;
  }
`;

export const MaterialRadio = styled(Radio)`
  svg {
    fill: ${theme.primary};
  }
`;
