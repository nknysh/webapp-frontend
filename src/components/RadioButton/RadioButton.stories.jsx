import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';

import RadioButton from './RadioButton';

storiesOf('RadioButton', module)
  .addDecorator(withKnobs)
  .add('default', () => {
    return <RadioButton options={[{ value: 'true', label: 'An option' }]} />;
  });
