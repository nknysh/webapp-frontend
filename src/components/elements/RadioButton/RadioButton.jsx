import React from 'react';

import { mapWithIndex } from 'utils';

import { propTypes, defaultProps } from './RadioButton.props';
import { StyledRadioGroup, RadioFormControl, MaterialRadio } from './RadioButton.styles';

const renderOption = (props, i) => <RadioFormControl key={i} {...props} control={<MaterialRadio />} />;

const RadioButton = ({ options, ...props }) => (
  <StyledRadioGroup {...props}>{mapWithIndex(renderOption, options)}</StyledRadioGroup>
);

RadioButton.propTypes = propTypes;
RadioButton.defaultProps = defaultProps;

export default RadioButton;
