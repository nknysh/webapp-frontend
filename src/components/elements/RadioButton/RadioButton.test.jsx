import React from 'react';

import RadioButton from './RadioButton';

const getComponent = props => shallow(<RadioButton {...props} />);

describe('<RadioButton />', () => {
  describe('render', () => {
    it('matches snapshots', () => {
      expect(getComponent({ options: [{ value: 'true', label: 'An option' }] })).toMatchSnapshot();
    });
  });
});
