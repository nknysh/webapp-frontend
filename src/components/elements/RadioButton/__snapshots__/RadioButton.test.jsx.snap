// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> render matches snapshots 1`] = `
<RadioButtonstyles__StyledRadioGroup>
  <RadioButtonstyles__RadioFormControl
    control={<ForwardRef />}
    key="0"
    label="An option"
    value="true"
  />
</RadioButtonstyles__StyledRadioGroup>
`;
